### DISCUSSION TITLE
I have some configuration in the .env.local file and i build the next app using 'npm run build' and start the app using 'next start' . Now I change some configuration in the .env.local file and restart the app without rebuilding . This is not picking my updated env file .Can I get some help here.

### DISCUSSION THREAD

**Discussion by BharathKS77** [vercel/next.js#41555] [Show and tell]
KEYCLOAK_ISSUER=http://localhost:8080/realms/new-app
KEYCLOAK_REALM_NAME=new-app
KEYCLOAK_ROLE_APPROVERS=approvers                    
KEYCLOAK_ROLE_OWNERS=owners                    
KEYCLOAK_BASE_URL=http://localhost:8080
NEXTAUTH_URL=http://localhost:3000
NEXTAUTH_SECRET=anyth;

**Comment by pchalupa** [comment]
Env variables are evaluated in build time. To see your updated env vars, you have to rebuild your app. See: https://nextjs.org/docs/basic-features/environment-variables#loading-environment-variables

**Comment by icyJoseph** [comment]
Hi,

This should work... for things that are not optimized/inlined at **build-time** that is.

For example if I use `getServerSideProps`:

```ts
export const getServerSideProps = () => {
  console.log(process.env.DB_URL);
  return { props: { foo: "bar" } };
};
```

When build and run, I can change DB_URL through .env files or any other way.

However if I had used `getStaticProps`, then that'll run with whatever value was available for it at build time, and it won't matter what I change at runtime. Unless I use ISR...

This is a very vanilla example, and this issue usually appears in non-obvious ways. How are these variables being used?
